### PR TITLE
Add support for multiple Wasm engines and wazero as the next implementation

### PR DESCRIPTION
**Pull request by pkedy**
Also adds support for TinyGo 0.22.0

/cc @madflojo

### CODE REVIEW

File: engine_test.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+			if err == nil {
+				t.Errorf("Expected error when creating module with invalid wasm, got nil")
+			}
+		})
+	}
+}
+
+func TestModule(t *testing.T) {
+	for _, engine := range engines {
+		t.Run(engine.Name(), func(t *testing.T) {
```
Comment (madflojo) [inline engine_test.go:117]: Should we wrap these tests in a guest for loop as well? 

```
		t.Run(engine.Name(), func(t *testing.T) {
			for l, p := range lang {
				t.Run("Module testing with "+l+" Guest", func(t *testing.T) {
```

Or alternatively, should we move some of these tests to the above Test?

**Comment by codefromthecrypt** [comment]
I will raise a PR against this branch for drift in wazero. We had a tentative API and the upcoming PR stabilizes!

**Comment by codefromthecrypt** [comment]
https://github.com/wapc/wapc-go/pull/15 ready for ya

File: engine_test.go (hunk 1 above)
Comment (pkedy) [reply engine_test.go:117]: Could make sense. I'm merging this PR now so @flavio can add wasmtime!

**Comment by codefromthecrypt** [comment]
thanks tons! I'll update this again shortly as we're nearly done clean resource management of WASI files.

**Pull request merged**